authorize_key table 에 idx 추가

key-del-jeeinho opened the pull request:
DataBase 수정
 authorize_key column 을 primary key 로 쓰던 기본 방식을 idx 라는 인조 키로 대체

key-del-jeeinho committed:
#authorize_key table 에 idx 추가
- authorize_key column 을 primary key 로 쓰던 기본 방식을 idx 라는 인조 키를 생성해 primary key 로 채택

siwony commented on src/main/java/com/promotion/devconsole/domain/converter/DeveloperDtoConverter.java:
```diff
@@ -24,7 +24,7 @@ public Developer convertToEntity(DeveloperDto dto) {
 
     @Override
     public DeveloperDto convertToDto(Developer entity) {
```
나중에 기회가 된다면 QueryDSL나 JPQL사용해보세요 DB에서 조회결과를 DTO로 바로 반환할 수 있습니다!

key-del-jeeinho replied:
한번 배워보고싶네요 유용한 정보 감사합니다 😁😁😁

siwony commented on src/main/java/com/promotion/devconsole/domain/converter/AuthorizeKeyDtoConverter.java:
```diff
@@ -19,6 +19,7 @@
     @Override
     public AuthorizeKey convertToEntity(AuthorizeKeyDto dto) {
```
DTO를 Entity로 변환하는 매서드 명은 주로 `toEntity`를 많이 사용하는거 같아요 참고만 해주세요

key-del-jeeinho replied:
알겠습니다 `toEntity` 와 `toDto` 로 수정하겠습니다~

siwony commented on src/main/java/com/promotion/devconsole/domain/entity/AuthorizeKey.java:
```diff
@@ -17,7 +17,11 @@
 
  private static final long serialVersionUID = 3601472106969795194L;
  @Id
+ @Column(name = "authorize_key")
  @GeneratedValue(strategy = GenerationType.IDENTITY)
```
`@Id`와 `@GeneratedValue`를 한줄로 하는것도 나쁘지 않아보여요

key-del-jeeinho replied:
넵 수정하겠습니다~ 😀

siwony reviewed (approved):
확인했습니다.

key-del-jeeinho committed:
#CodeReview 수용
## Converter 내 메서드 네이밍 수정
convertToEntity -> toEntity
convertToDto -> toDto
## 가독성 증가를 위한 Dto 내 Annotation 배치 변경
Id Annotation 과 GeneratedValue Annotation 을 한줄에 배치
